src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      const index = Animal.alive.indexOf(animal);
+
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
Instead of checking the animal's health and then using `splice` to remove it from `Animal.alive`, it's preferred to use the `filter` method to remove animals with health less than or equal to 0. This approach is more concise and aligns with the preferred method of handling the `Animal.alive` array.